ui/src/containers/NavigationModalContainer.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { Button } from '@patternfly/react-core';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { toggleNavigationModal, toggleAppDetailedIsDirty } from '../actions/actions-ui';
+import { LargeModal } from '../components/common/LargeModal';
+
+class NavigationModalContainer extends React.Component {
+  handleModalClose = () => {
+    this.props.toggleNavigationModal(false);
+  };
+
+  handleLeaveClick = () => {
+    this.props.toggleAppDetailedIsDirty();
+    this.props.history.goBack();
+    this.handleModalClose();
+  };
+
+  render () {
+    return (
+      <LargeModal
+        title="Are you sure you want to leave this page?"
```

Comment:
I am in favour of making components as reusable as possible. To make this component reusable we can pass props that are likely to be different on different pages.

The following could be passes in as `props`:

* title
* text
* Primary button text (Leave)
* Secondary button (Stay)

Reply:
You believe the text should be stored in the redux state and passed into this component?

Reply:
You don't have to store something in the state in order to pass it as a prop to a component. You can still pass props to your component, like this:

```jsx
return (
    <NavigationModalContainer title="Are you sure you want to leave this page?"/>
);
```

Reply:
Changed now for the title and text. I don't believe the button text should change as the handlers are very specific to the what they should be doing.